internal/bminventory/inventory.go

```diff
@@ -4123,10 +4167,14 @@ func (b *bareMetalInventory) UpdateInfraEnvInternal(ctx context.Context, params
 				errWrapperLog = log.WithError(err)
 			}
 			errWrapperLog.Errorf("Failed to update InfraEnv with id %s", params.InfraEnvID)
+			tx.Rollback()
+			if r := recover(); r != nil {
+				tx.Rollback()
+			}
 		}
 	}()
 
-	if infraEnv, err = common.GetInfraEnvFromDB(b.db, params.InfraEnvID); err != nil {
+	if infraEnv, err = common.GetInfraEnvFromDB(tx, params.InfraEnvID); err != nil {
```

Comment:
I don't think so since this object is [passed](https://github.com/openshift/assisted-service/pull/3791/files#diff-747a429a0e12307ef65d754c83b6898ce3858b621005e1e7af6e420da1f1657cR4201) to the [`updateInfraEnvData`](https://github.com/openshift/assisted-service/blob/7b0200234511254c2315eed7490a0c39fbbd676b/internal/bminventory/inventory.go#L4222) function which only updates certain columns instead of the full object